handlers.go

```diff
@@ -4215,7 +4218,8 @@ func (s *Server) handleColfsListDir(c echo.Context, u *User) error {
 
 		if !strings.Contains(relp, "/") {
 			out = append(out, collectionListResponse{
-				Name:   filepath.Base(relp),
+				// Name:   filepath.Base(relp),
+				Name:   *r.Filename,
```

Comment:
do we have confidence this will never be nil?

Reply:
Good one! In https://github.com/application-research/estuary/blob/67fce679308c584eef3aa5833ff426ae8b429f40/handlers.go#L476-L479 there's a check for the empty string, I assumed that if you don't specify a `name` field on the request it gets set as an empty string. Is that assumption right?

Reply:
FWIW, this is more a question about the database than the request handler. if we have any database rows with a `nil` Filename. maybe to protect against that we can just add a check on line 4196, i.e...

```go
		if r.Path == nil || r.Filename == nil {
				continue
		}
```

Reply:
Oh I see! Makes sense